### PR TITLE
chore, fix(egen): restructuring from template, markdown corrections & dataset object fix

### PR DESCRIPTION
**Pull request by krishr2d2**
<br>
This PR includes the following updates:

1. Restructures the sections according to the template. Updates the Colab related markdown and cells accordingly.
2. Reduces the usage of future tense.
3. Adds the collapsed license comment.
4. removes IS_TESTING parameter in cleaning up section.
5. Converts Dataset object to a list while passing the object to the Vector search endpoint and fixes the prediction issue.
<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

### CODE REVIEW

File: notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb [hunk 1]
```diff
@@ -30,23 +32,27 @@
...
+        "      <img src=\"https://cloud.google.com/ml-engine/images/colab-logo-32px.png\" alt=\"Google Colaboratory logo\"><br> Open in Colab\n",
         "    </a>\n",
         "  </td>\n",
-        "  <td>\n",
-        "    <a href=\"https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb\">\n",
-        "      <img src=\"https://cloud.google.com/ml-engine/images/github-logo-32px.png\" alt=\"GitHub logo\">\n",
-        "      View on GitHub\n",
+        "  <td style=\"text-align: center\">\n",
+        "    <a href=\"https://console.cloud.google.com/vertex-ai/colab/import/https:%2F%2Fraw.githubusercontent.com%2FGoogleCloudPlatform%2Fvertex-ai-samples%2Fmain%2Fnotebooks%2Fofficial%2Fmatching_engine%2Fsdk_matching_engine_for_indexing.ipynb\">\n",
+        "      <img width=\"32px\" src=\"https://lh3.googleusercontent.com/JmcxdQi-qOpctIvWKgPtrzZdJJK-J3sWE1RsfjZNwshCFgE_9fULcNpuXYTilIR2hjwN\" alt=\"Google Cloud Colab Enterprise logo\"><br> Open in Colab Enterprise\n",
```
Comment (kittyabs) [inline notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb:45]: Update colab enterprise URL:

-https://cloud.google.com/ml-engine/images/colab-enterprise-logo-32px.png

File: notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb [hunk 2]
```diff
@@ -101,15 +107,22 @@
...
     {
       "cell_type": "markdown",
       "metadata": {
         "id": "f0f1bea346db"
       },
       "source": [
-        "## Installation\n",
-        "\n",
-        "Install the latest versions of packages required to execute this notebook."
+        "### Install Vertex AI SDK and other required packages\n"
```
Comment (kittyabs) [inline notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb:125]: Vertex AI SDK for Python

File: notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb [hunk 3]
```diff
@@ -144,134 +159,75 @@
...
-      "execution_count": null,
-      "metadata": {
-        "id": "474be5183c27"
+        "id": "7b49231643e4"
       },
-      "outputs": [],
       "source": [
-        "REGION = \"us-central1\"  # @param {type: \"string\"}"
+        "<div class=\"alert alert-block alert-warning\">\n",
+        "<b>⚠️ The kernel is going to restart. Please wait until it is finished before continuing to the next step. ⚠️</b>\n",
```
Comment (kittyabs) [inline notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb:179]: Remove "Please" and contract "it is"

- Wait until it's finished...

File: notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb [hunk 4]
```diff
@@ -4,10 +4,12 @@
       "cell_type": "code",
       "execution_count": null,
       "metadata": {
+        "cellView": "form",
         "id": "ur8xi4C7S06n"
       },
       "outputs": [],
       "source": [
+        "# @title Copyright & License (click to expand)\n",
```
Comment (katiemn) [inline notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb:12]: Should this be removed?

File: notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb [hunk 5]
```diff
@@ -101,15 +107,22 @@
         "\"GloVe is an unsupervised learning algorithm for obtaining vector representations for words. Training is performed on aggregated global word-word co-occurrence statistics from a corpus. The resulting representations showcase interesting linear substructures of the word vector space.\"\n"
       ]
     },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "8925ff9e165e"
+      },
+      "source": [
+        "## Getting Started"
```
Comment (katiemn) [inline notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb:116]: "Getting Started" -> "Get started"

File: notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb [hunk 6]
```diff
@@ -346,36 +302,35 @@
...
-        "  * **In the same region as where your ANN service is deployed** (for example, if you set `REGION = \"us-central1\"` as same as the tutorial, the notebook instance has to be in `us-central1`).\n",
-        "  * **Make sure you select the VPC network you created for ANN service** (instead of using the \"default\" one). That is, you have to create the VPC network below and then create a new notebook instance that uses that VPC.  \n",
-        "  * If you run it in the colab or a Vertex AI Workbench notebook instance in a different VPC network or region, \"Create Online Queries\" section will fail."
+        "**WARNING:** The `MatchingIndexEndpoint.match` method (to create online queries against your deployed index) has to be executed in a Vertex AI Workbench notebook instance that is created with the following requirements:\n",
+        "  * **In the same region as where your ANN service is deployed** (for example, if you set `LOCATION = \"us-central1\"` as same as the tutorial, the notebook instance has to be in `us-central1`).\n",
```
Comment (katiemn) [inline notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb:306]: Possibly reword to: **In the same region where your ANN service is deployed** (for example, if you set `LOCATION = \"us-central1\"`, the notebook instance has to be in `us-central1`).

File: notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb [hunk 7]
```diff
@@ -346,36 +302,35 @@
...
+        "  * **Make sure you select the VPC network you created for ANN service** (instead of using the \"default\" one). That is, you have to create a new notebook instance that uses the VPC network you created earlier. Run the rest of the tutorial from that notebook instance.\n",
+        "  * If you run it in the colab or a Vertex AI Workbench notebook instance in a different VPC network or region, \"Create Online Queries\" section is expected to fail."
```
Comment (katiemn) [inline notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb:308]: Possibly modify to: If you run it in a Colab or Vertex AI Workbench notebook in a different VPC network or region, the \"Create Online Queries\" section is expected to fail.

File: notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb (hunk 4 above)
Comment (krishr2d2) [reply notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb:12]: The collapsed license was added as it was listed in the [template guidelines](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template_writing_guidelines.md).

Comment (krishr2d2) [reply notebooks/official/matching_engine/sdk_matching_engine_for_indexing.ipynb]: Removed the collapsed comment.

**Review by katiemn — APPROVED** [review]

**Pull request merged**